packages/teleterm/src/ui/TopBar/Connections/ConnectionsFilterableList/ConnectionItem.tsx

```diff
@@ -48,37 +48,43 @@ export function ConnectionItem(props: ConnectionItemProps) {
         height: unset;
       `}
     >
-      <ConnectionStatusIndicator mr={3} connected={props.item.connected} />
+      <ConnectionStatusIndicator
+        mr={3}
+        css={`
+          flex-shrink: 0;
+        `}
+        connected={props.item.connected}
+      />
       <Flex
         alignItems="center"
         justifyContent="space-between"
         flex="1"
-        width="100%"
         minWidth="0"
       >
-        <Flex flexDirection="column">
+        <div
+          css={`
+            min-width: 0;
+          `}
+        >
           <Text
             typography="body1"
             bold
             color="text.primary"
             title={props.item.title}
             css={`
               line-height: 16px;
-              white-space: normal;
```

Comment:
`white-space: normal` was causing the UI to behave inconsistently. Document titles or cluster names with hyphens in them were getting split on multiple lines with no ellipsis, while the same elements without hyphens were cut and an ellipsis was added.